File: BackEnd/src/main/java/hector/vico/EntreBicis/entity/EstatRecompensa.java
package hector.vico.EntreBicis.entity;

/**
 * Enum que representa els possibles estats d'una recompensa dins del sistema EntreBicis.
 * 
 * <ul>
 *   <li><b>DISPONIBLE:</b> Recompensa visible i pot ser reservada per un usuari.</li>
 *   <li><b>RESERVADA:</b> Ja ha estat reservada per un usuari, però encara no assignada.</li>
 *   <li><b>ASSIGNADA:</b> Confirmació que la recompensa ha estat aprovada per l'equip i està pendent de recollir.</li>
 *   <li><b>RECOLLIDA:</b> L'usuari ha recollit físicament la recompensa.</li>
 * </ul>
 */
public enum EstatRecompensa {
    DISPONIBLE,  // La recompensa està lliure per ser reservada
    RESERVADA,   // Ja ha estat reservada per un usuari
    ASSIGNADA,   // Ha estat assignada de manera oficial a un usuari
    RECOLLIDA    // L'usuari ha recollit la recompensa
}

File: FrontEnd/app/src/main/java/cat/copernic/hvico/entrebicis/Core/Model/EstatRecompensa.kt
package cat.copernic.hvico.entrebicis.Core.Model

/**
 * Enum que representa els diferents estats que pot tenir una recompensa.
 *
 * @property DISPONIBLE La recompensa encara no ha estat reservada per cap usuari.
 * @property RESERVADA La recompensa ha estat reservada per un usuari però encara no assignada.
 * @property ASSIGNADA La recompensa ha estat validada i assignada a l'usuari.
 * @property RECOLLIDA La recompensa ja ha estat recollida físicament per l'usuari.
 */
enum class EstatRecompensa {
    DISPONIBLE,
    RESERVADA,
    ASSIGNADA,
    RECOLLIDA
}